src/fdcache_entity.cpp

```diff
@@ -2561,6 +2581,8 @@ bool FdEntity::GetLastUpdateUntreatedPart(off_t& start, off_t& size) const
 
 bool FdEntity::ReplaceLastUpdateUntreatedPart(off_t front_start, off_t front_size, off_t behind_start, off_t behind_size)
 {
+    const std::lock_guard<std::mutex> lock(fdent_lock);
```

Comment:
This method is called as a callback from PseudoFdInfo::UploadBoundaryLastUntreatedArea().
The FdEntity object passed to PseudoFdInfo::UploadBoundaryLastUntreatedArea() has already performed this locking, so it cannot be locked here.